dask_kubernetes/core.py

```diff
@@ -182,67 +410,69 @@ def __init__(
             )
             raise ValueError(msg)
 
-        pod_template = clean_pod_template(pod_template)
-        ClusterAuth.load_first(auth)
+        self.pod_template = clean_pod_template(self.pod_template)
+        await ClusterAuth.load_first(self.auth)
 
         self.core_api = kubernetes.client.CoreV1Api()
 
-        if namespace is None:
-            namespace = _namespace_default()
+        if self._namespace is None:
+            self._namespace = _namespace_default()
 
-        name = name.format(
+        self._generate_name = self._generate_name.format(
             user=getpass.getuser(), uuid=str(uuid.uuid4())[:10], **os.environ
         )
-        name = escape(name)
-        self.pod_template = pod_template
+        self._generate_name = escape(self._generate_name)
 
         # Default labels that can't be overwritten
-        self.pod_template.metadata.labels["dask.org/cluster-name"] = name
+        self.pod_template.metadata.labels["dask.org/cluster-name"] = self._generate_name
         self.pod_template.metadata.labels["user"] = escape(getpass.getuser())
         self.pod_template.metadata.labels["app"] = "dask"
-        self.pod_template.metadata.labels["component"] = "dask-worker"
-        self.pod_template.metadata.namespace = namespace
-
-        self.cluster = LocalCluster(
-            host=host or socket.gethostname(),
-            scheduler_port=port,
-            n_workers=0,
-            **kwargs
-        )
+        self.pod_template.metadata.namespace = self._namespace
 
-        # TODO: handle any exceptions here, ensure self.cluster is properly
-        # cleaned up.
-        self.pod_template.spec.containers[0].env.append(
-            kubernetes.client.V1EnvVar(
-                name="DASK_SCHEDULER_ADDRESS", value=self.scheduler_address
-            )
-        )
-        if env:
+        if self.env:
             self.pod_template.spec.containers[0].env.extend(
                 [
                     kubernetes.client.V1EnvVar(name=k, value=str(v))
-                    for k, v in env.items()
+                    for k, v in self.env.items()
                 ]
             )
-        self.pod_template.metadata.generate_name = name
+        self.pod_template.metadata.generate_name = self._generate_name
 
-        finalize(self, _cleanup_pods, self.namespace, self.pod_template.metadata.labels)
+        finalize(
+            self, _cleanup_resources, self._namespace, self.pod_template.metadata.labels
+        )
 
-        if n_workers:
-            try:
-                self.scale(n_workers)
-            except Exception:
-                self.cluster.close()
-                raise
+        common_options = {
+            "core_api": self.core_api,
+            "pod_template": self.pod_template,
+            "namespace": self._namespace,
+            "loop": self.loop,
+        }
+
+        if self._local_scheduler:
+            self.scheduler_spec = {"cls": LocalScheduler, "options": {}}
```

Comment:
It might be important to pass through some minimal set of options here.  This is what we use for dask-jobqueue:

```python
        scheduler = {
            "cls": Scheduler,  # Use local scheduler for now
            "options": {
                "protocol": protocol,
                "interface": interface,
                "host": host,
                "dashboard_address": dashboard_address,
                "security": security,
            },
        }
```